02/app.js

```diff
@@ -0,0 +1,78 @@
+const formEl = document.querySelector('form')
+const emailInputEl = document.getElementById("formLogin");
+const passwordInputEl = document.getElementById("formPass1");
+const secondPasswordInputEl = document.getElementById("formPass2");
+const checkBoxEl = document.getElementById("formAccept");
+
+const errors = [];
+
+//walidacja mail:
+
+const emailValidation = () => {
+ const emailValue = emailInputEl.value
+ 
+ const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+ if(!emailRegex.test(emailValue)){
+    errors.push('#formLogin')
+ }
+}
+
+//walidacja hasła:
+const firstPasswordValidation = () =>{
+    const passwordValue = passwordInputEl.value;
+    if(passwordValue.length <= 6){
+        errors.push("#formPass1")
+    }
+}
+
+const secondPasswordValidation = () =>{
+    const passwordValue = passwordInputEl.value;
+    const secondPasswordValue = secondPasswordInputEl.value
+    if(passwordValue !== secondPasswordValue){
+        errors.push("#formPass2");
+    }
+}
+
+const regulationsValidation = () => {
+ if(!checkBoxEl.checked){
+    errors.push("#formAccept");
+ }
+}
+
+const handleError = () => {
+	// Zresetuj style przed sprawdzaniem błędów
+	document.querySelectorAll("form label").forEach((label) => {
+		label.style.color = ""; // Zresetuj kolor
+	});
+
+	// Oznacz błędne etykiety na czerwono
+	errors.forEach((selector) => {
+		const label = document.querySelector(`${selector} + label`);
```

Comment:
Zwróć uwagę, że relacja w HTML elementów `labe`l oraz `input` jest inna niż ten zapis (`label` nie jest następnym element po `input` - tylko odwrotnie), tam mamy:
```
            <label for="formLogin">
                email:
            </label>
            <input id="formLogin" name="login" type="email" /> 
```

Dlatego to wyszukanie powinno być takie:
 ```const label = document.querySelector(`${selector}`).parentElement.querySelector('label');```